components/core/src/compressor_frontend/InputBuffer.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef COMPRESSOR_FRONTEND_INPUT_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_INPUT_BUFFER_HPP
+
+// Project Headers
+#include "../ReaderInterface.hpp"
+#include "Buffer.hpp"
+
+namespace compressor_frontend {
+    class InputBuffer : public Buffer<char> {
```

Comment:
* Missing class description.
* Seeing as the code seems to assume a very specific use, I would change the class name to something like `InputCharacterBuffer` or `LogMessageInputBuffer`.